src/plugins/template/tests/functional/op_reference/unique.cpp

```diff
@@ -239,103 +246,104 @@ std::vector<UniqueParams> params_unique_int() {
+                     std::vector<Data_t>{6, 5, 4, 3, 2, 1},
+                     std::vector<Index_t>{0},
+                     std::vector<Index_t>{0, 0},
+                     std::vector<Count_t>{2},
+                     make_axis(1),
+                     false,
+                     "3D with duplicates"},
+        UniqueParams{Shape{2, 2, 3},
+                     // the first and the last slice over axis 2 are equal
+                     std::vector<Data_t>{-1, 2, -1, 5, -3, 5, 7, -8, 7, 4, 4, 4},
+                     std::vector<Data_t>{-1, 2, 5, -3, 7, -8, 4, 4},
+                     std::vector<Index_t>{0, 1},
+                     std::vector<Index_t>{0, 1, 0},
+                     std::vector<Count_t>{2, 1},
+                     make_axis(2),
+                     false,
+                     "3D with duplicates(1 & 3)"},
+        UniqueParams{Shape{2, 2, 3},
+                     // the first and the second slice over axis 2 are equal
+                     std::vector<Data_t>{-1, -1, 2, 5, 5, -3, 7, 7, -8, 4, 4, 4},
+                     std::vector<Data_t>{-1, 2, 5, -3, 7, -8, 4, 4},
+                     std::vector<Index_t>{0, 2},
+                     std::vector<Index_t>{0, 0, 1},
+                     std::vector<Count_t>{2, 1},
+                     make_axis(2),
+                     false,
+                     "3D with duplicates (1 & 2)"},
+        UniqueParams{Shape{2, 2, 3},
+                     // the second and the third slice over axis 2 are equal
+                     std::vector<Data_t>{2, -1, -1, -3, 5, 5, -8, 7, 7, 4, 4, 4},
+                     std::vector<Data_t>{2, -1, -3, 5, -8, 7, 4, 4},
+                     std::vector<Index_t>{0, 1},
+                     std::vector<Index_t>{0, 1, 1},
+                     std::vector<Count_t>{1, 2},
+                     make_axis(2),
+                     false,
+                     "3D with duplicates (2 & 3)"},
+        UniqueParams{Shape{2, 2, 3},
+                     // the second and the third slice over axis 2 are equal
+                     std::vector<Data_t>{2, -1, -1, -3, 5, 5, -8, 7, 7, 4, 4, 4},
+                     std::vector<Data_t>{-1, 2, 5, -3, 7, -8, 4, 4},
+                     std::vector<Index_t>{1, 0},
+                     std::vector<Index_t>{1, 0, 0},
+                     std::vector<Count_t>{2, 1},
+                     make_axis(2),
+                     true,
+                     "3D with duplicates (2 & 3), output sorted"},
+        UniqueParams{Shape{2, 2, 3},
+                     // the second and the third slice over axis 2 are equal
+                     std::vector<Data_t>{-1, -1, -1, 3, 2, 2, 6, 7, 7, 4, 4, 4},
+                     std::vector<Data_t>{-1, -1, 2, 3, 7, 6, 4, 4},
+                     std::vector<Index_t>{1, 0},
+                     std::vector<Index_t>{1, 0, 0},
+                     std::vector<Count_t>{2, 1},
+                     make_axis(2),
```

Comment:
Are there tests for negative axis?
https://github.com/openvinotoolkit/openvino/pull/14250/files#diff-85271b9468a0003ae1c1d0d40a600bc24da69fa968592bcc130a972d6277941dR116 - here is rough fix for axes in reference. Could you please add tests for that?

Reply:
I've added the tests and handling of the negative axis in the reference implementation. Since the axis is held by a unique_ptr by the reference implementation you can safely modify it inside so please consider using this approach in your PR https://github.com/openvinotoolkit/openvino/pull/14315/files#diff-85271b9468a0003ae1c1d0d40a600bc24da69fa968592bcc130a972d6277941dR111-R114